search.ipynb

```diff
@@ -1044,6 +770,87 @@
     "* `search(self, problem)`: This method is used to search a sequence of `actions` to solve a `problem`."
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let us now define a Simple Problem Solving Agent Program. We will create a simpleAgent class which will inherit from the abstract class `SimpleProblemSolvingAgentProgram` and overides it methods. We will create a simple intelligent vacuum agent which can be in any one of the following states. It will move to any other state depending upon the current state as shown in the picture by arrows:\n",
```

Comment:
I am not sure what 'simpleAgent' is. If it is two words, you should add a space between them. If it is a var/class name, you should put  `  around it.

'Overide' has two 'r's. Also, you write 'overide it methods', while it should be 'its' instead of 'it'.